src/js/components/Filter/FilterForm.jsx

```diff
@@ -31,21 +33,18 @@ const FilterForm = ({
     },
   };
   // Create initialValues from filterFields as empty values
-  const initialValues = Object.keys(filterFields).reduce((acc, curr) => {
-    if (!acc[curr]) {
-      return {
-        ...acc,
-        [curr]: '',
-      };
-    }
-    return acc;
-  }, { ...defaultValues });
-  const [amountFilled, setAmountFilled] = useState(0);
+  let initialValues = Object.keys(filterFields)
+    .reduce((acc, key) => ({ ...acc, [`${key}`]: null }), {});
+  initialValues = { ...initialValues, ...defaultValues };
```

Comment:
It is a very simple code and basically works exactly the same as the original by initially setting `null` values to all of the filters and after overwriting them with `defaultValues` if the exist .
It's just in this case I saw the opportunity to make this code shorter without any `if(...)` statements.
But I see your point and I don't have any problem with bringing back the previous version.

Reply:
my bad I haven't told you what the bug might be about - the problem is that you changed `''` to `null` - there is one case in PO list where I was checking if one of params was === `''` (couldn't do it any other way) so changing from `''` to `null` would break this case there.
We can agree that this `if` is for 0.0001% as the situation with two the same keys would probably never happen, so I would meet you halfway if you at least made the default `acc` as `{...defaultValues}`, because `.reduce` allows that and there is no point of using empty acc and having to use `let` to be able to assign the `defualtValues` then.